pybamm/install_odes.py

```diff
@@ -75,40 +76,36 @@ def install_sundials(download_dir, install_dir):
     make_cmd = ["make", "install"]
     subprocess.run(make_cmd, cwd=build_directory, check=True)
 
-
 def update_LD_LIBRARY_PATH(install_dir):
-    # Look for current python virtual env and add export statement
-    # for LD_LIBRARY_PATH in activate script.  If no virtual env found,
-    # then the current user's .bashrc file is modified instead.
+    # Look for the current python virtual env and add an export statement
+    # for LD_LIBRARY_PATH in the activate script. If no virtual env is found,
+    # the current user's .bashrc file is modified instead.
 
-    export_statement = "export LD_LIBRARY_PATH={}/lib:$LD_LIBRARY_PATH".format(
-        install_dir
-    )
+    export_statement = f"export LD_LIBRARY_PATH={install_dir}/lib:$LD_LIBRARY_PATH"
 
     venv_path = os.environ.get("VIRTUAL_ENV")
     if venv_path:
         script_path = os.path.join(venv_path, "bin/activate")
     else:
-        script_path = os.path.join(os.environ.get("HOME"), ".bashrc")
-
-    if os.getenv("LD_LIBRARY_PATH") and "{}/lib".format(install_dir) in os.getenv(
-        "LD_LIBRARY_PATH"
-    ):
-        print("{}/lib was found in LD_LIBRARY_PATH.".format(install_dir))
-        print("--> Not updating venv activate or .bashrc scripts")
+        if sys.platform == "linux":
+            script_path = os.path.join(os.environ.get("HOME"), ".bashrc")
+        if sys.platform == "darwin":
+            script_path = os.path.join(os.environ.get("HOME"), ".zshrc")
+
```

Comment:
Not sure if this will factor in the cases for those who don't use `zsh` as the default shell. Is there an elegant way to detect the default terminal shell programmatically and append the installation directory to LD_LIBRARY_PATH in a more Pythonic way (which is also permanent)? I haven't dived into the internals of the `os` module that deep but I think it should be possible. 

That level of abstraction will clean up some bits of the script which are opening and writing to these shell configuration files. Users will then not be restricted to using `bash` or `zsh`.

Reply:
Detecting default shell type via `os.environ` now.